protos/orchestrator_service.proto

```diff
@@ -376,6 +376,124 @@ message DeleteTaskHubResponse {
     //no playload
 }
 
+message SignalEntityRequest {
+    string instanceId = 1;
+    string name = 2;
+    google.protobuf.StringValue input = 3;
+    bytes guid = 4;
+    bool hasScheduledTime = 5;
+    google.protobuf.Timestamp scheduledTime = 6;
+}
+
+message SignalEntityResponse {
+     // no payload
+}
+
+message GetEntityRequest
+{
+    string instanceId = 1;
+    bool includeState = 2;
+}
+
+message GetEntityResponse
+{
+    bool exists = 1;
+    EntityMetadata entity = 2;
+}
+
+message EntityQuery
+{
+    google.protobuf.StringValue instanceIdStartsWith = 1;
+    google.protobuf.Timestamp lastModifiedFrom = 2;
+    google.protobuf.Timestamp lastModifiedTo = 3;
+    bool includeState = 4;
+    int32 pageSize = 5;
+    google.protobuf.StringValue continuationToken = 6;
+}
+
+message QueryEntitiesRequest
+{
+    EntityQuery query = 1;
+}
+
+message QueryEntitiesResponse
+{
+    repeated EntityMetadata entities = 1;
+    google.protobuf.StringValue continuationToken = 2;
+}
+
+message EntityMetadata
+{
+    string instanceId = 1;
+    google.protobuf.Timestamp lastModifiedTime = 2;
+    string serializedState = 3;
+}
+
+message CleanEntityStorageRequest
+{
+    bool removeEmptyEntities = 1;
+    bool releaseOrphanedLocks = 2;
+}
+
+message CleanEntityStorageResponse
+{
+    int32 emptyEntitiesRemoved = 1;
+    int32 orphanedLocksReleased = 2;
+}
+
+message EntityBatchRequest {
+    string instanceId = 1;
+    google.protobuf.StringValue entityState = 2;
+    repeated OperationRequest operations = 3;
+}
+
+message EntityBackendProperties
+{
+    int64 entityMessageReorderWindow = 1;
+    int64 maximumSignalDelayTime = 2;
+}
+
+message EntityBatchResult {
+    repeated OperationResult results = 1;
+    repeated OperationAction actions = 2;
+    google.protobuf.StringValue entityState = 3;
+}
+
+message OperationRequest {
+    string operation = 1;
+    bytes guid = 2;
+    google.protobuf.StringValue input = 3;
+}
+
+message OperationResult {
+    google.protobuf.StringValue result = 1;
+    google.protobuf.StringValue errorCategory = 2;
```

Comment:
What is `errorCategory`?

Reply:
the meaning of the fields in `OperationResult` is a bit tricky since this is a data structure defined in DT.Core and its interpretation depends on the SDK used (e.g. quite different meanings for DF legacy). The only invariant is that a null value indicates "no error" and any non-null value indicates "error".

For Isolated, I use this field to contain a short message indicating where the error was caught, and then use the `TaskFailureDetails` to contain the full error description.

Reply:
I am not sure I see a big benefit to including this and it may be confusing. Can we leave it out until we see tangible benefits to it? Offering a little bit of human readable info on where it was caught doesn't seem like it is worth including this. Have to maintain it, customers may be confused by it (why two error related properties?), and the serialization overhead.

I am fine leaving it in the DT.Core SDK, but not sure we need to expose it here.

Reply:
the problem is that DT.Core.Entities.Operation is still being used by the front ends, so if we remove it from this proto then it means the serialization/deserialization is not consistent, i.e. an OperationResult serialized in one place is not guaranteed to be deserialized to the same OperationResult in another.

Reply:
> Offering a little bit of human readable info on where it was caught doesn't seem like it is worth including this

I agree that having two fields is kind of dumb (it is a compatibility thing, not really what I would have picked otherwise). That said, keeping the information about where the error originates from is essential IMO. There are so many different places, in different dlls, that can generate errors.

If we use only FailureDetails then we would have to wrap them in order to keep that information. I can make that change.

Reply:
One more complication here. The problem is that protobuf has no null. So, it is always going to deserialize to a non-null FailureDetails. So I also have to add a boolean 'isErrorResult' to indicate whether this is a failed result or not.

Reply:
I have revised the encoding for operation results.